src/mongocxx/validation_criteria.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mongocxx/validation_criteria.hpp>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+void validation_criteria::rule(bsoncxx::document::value rule) {
+    _owned_rule = std::move(rule);
+    _rule = _owned_rule->view();
+}
+
+void validation_criteria::rule(bsoncxx::document::view rule) {
+    _rule = std::move(rule);
+    if (_owned_rule) {
+        _owned_rule = stdx::nullopt;
+    }
+}
+
+void validation_criteria::level(stdx::string_view level) {
+    _level = std::move(level);
+}
+
+void validation_criteria::action(stdx::string_view action) {
+    _action = std::move(action);
+}
+
+bsoncxx::document::value validation_criteria::to_document() const {
+    auto doc = bsoncxx::builder::stream::document{};
+
+    if (_rule) {
+        doc << "validator" << bsoncxx::builder::stream::open_document
```

Comment:
Does `doc << "validator" << *_rule;` work? I want there to be a better way to append a document than new doc, concat, close doc.